omisego-sdk/src/main/java/co/omisego/omisego/websocket/interval/SocketHeartbeat.kt

```diff
@@ -38,19 +39,26 @@ class SocketHeartbeat(
      * @param task A lambda with a [SocketSend] parameter. This will be executed periodically that starts immediately.
      */
     @Suppress("OVERRIDE_BY_INLINE")
+    @Synchronized
     override inline fun startInterval(crossinline task: (SocketSend) -> Unit) {
-        timer = Timer()
-        timer?.schedule(Date(), period, {
-            task(SocketSend(EVENT_NAME, SocketEventSend.HEARTBEAT, socketMessageRef.value, mapOf()))
-        })
+        synchronized(this) {
+            timer?.cancel()
+            timer = Timer()
```

Comment:
For debugging purpose, maybe you should specify a name to the timer? 
see https://docs.oracle.com/javase/7/docs/api/java/util/Timer.html#Timer(java.lang.String)

Btw, you could also pass `true` to isDaemon parameter, this way, this thread won't be blocking the process at any time. What do you think?

Reply:
Btw, maybe you could store the task, and cancel the current task if any here instead of cancelling the Timer and creating a new one. This would avoid the need to destroy/create a new thread, which is not free.
Plus, by doing this you can probably make the timer lazy and move the declaration, and you'll have a smaller synchronized block, which is good :)